data_adapter/structure.py

```diff
@@ -150,3 +152,149 @@ def _init_parameters(self, parameter_sheet: str) -> dict:
+                            [commodity for commodity in outputs if commodity.startswith(category)]
+                        )
+
+            # Create the matrix visualization for the current category
+            combined = {sector: list(input_commodities[sector] | output_commodities[sector]) for sector in sectors}
+            unique_commodity_list = sorted(list(set(commodity for sector in combined.values() for commodity in sector)))
+
+            # Create a DataFrame with initial values of 0
+            matrix_data = pd.DataFrame(np.NaN, index=unique_commodity_list, columns=sectors)
+
+            # Update the DataFrame with -1 for input, 1 for output, and 0 for both
+            for sector in sectors:
+                for commodity in combined[sector]:
+                    if commodity in input_commodities[sector] and commodity in output_commodities[sector]:
+                        matrix_data.at[commodity, sector] = 0
+                    elif commodity in input_commodities[sector]:
+                        matrix_data.at[commodity, sector] = -1
+                    elif commodity in output_commodities[sector]:
+                        matrix_data.at[commodity, sector] = 1
+
+            # Plot the matric information
+            ax = axes[i]
+            ax.imshow(matrix_data.values, cmap="RdYlGn", vmin=-1, vmax=1)
+
+            # Create a frame around each cell in the matrix for better readability
+            for j in range(len(matrix_data.index)):
+                for k in range(len(matrix_data.columns)):
+                    value = matrix_data.values[j, k]
+                    color = (
+                        "white" if np.isnan(value) else "red" if value == -1 else "green" if value == 1 else "yellow"
+                    )
+                    rect = plt.Rectangle(
+                        (k - 0.5, j - 0.5), 1, 1, fill=True, facecolor=color, edgecolor="black", linewidth=1
+                    )
+                    ax.add_patch(rect)
+
+            # Labeling
+            plt.sca(axes[i])
+            ax.set_xticks(range(len(matrix_data.columns)), matrix_data.columns)
+            ax.set_yticks(range(len(matrix_data.index)), matrix_data.index)
+            ax.set_xlabel("Sectors", fontsize=12)
+            ax.set_title(category.upper(), fontsize=16)
+
+        # Create a legend for the colors
+        legend_elements = [
+            Patch(facecolor="white", edgecolor="black", label="No Relation"),
+            Patch(facecolor="red", edgecolor="black", label="Input"),
+            Patch(facecolor="green", edgecolor="black", label="Output"),
+            Patch(facecolor="yellow", edgecolor="black", label="Input & Output"),
+        ]
+        fig.legend(handles=legend_elements, loc="upper left", fontsize=12)
+        plt.tight_layout()
+
+        # Save the plot as svg
+        plt.show()
+
+    def get_commodity_diff(self, input_processes: list = ["source", "import"], output_processes: list = ["sink"]):
```

Comment:
One question: Do we need to set input and output processes explicitly? I would think, that processes which either have only inputs or only outputs are "creating" or "destroying" commodities implicitly - that would it make easier in case some processes are not named sink/source or import...

Reply:
Havent thought it that way, thank you!

Reply:
Maybe it would also make sense to have an additional color (e.g. blue) for commodities that only come from  imports/sources. I think for the interface check this might be an interesting value.

sinks and exports could be filtered as mentioned above. However, sources and imports should be checked. I accordingly adjust my proposed filter in the review below.

Reply:
I have added an additional color. (picture below)

Reply:
This function is also separated from the function for drawing the commodities. 
It is meant only to get a list of "open" commodities, therefore I would like to include sources and sinks (in opposite of my thinking above) to see what commodities are either not destroyed or not created yet.

Reply:
I am not sure if this really helps, as now all these open commodities are blue and we do not know if it is an input or output, right? We could see those processes before, if one line did only have green or red cell.
The idea of the additional color was more to highlight commodities which have no other upstream processes except the imports/sources.

If you want it more clear that there is something "open" you could maybe use another formating. Like highighting the text or something?

Reply:
Yes we did know what inputs/outputs were but I did not understand correctly what you wanted :)
So here is the improved version for the Steel Industry Part O1: 
![Figure_1](https://github.com/sedos-project/data_adapter/assets/98911066/9da67248-b5fe-4abd-abf6-d0572944a6b3)